master/networking/assign-ip-addresses-topology.md

```diff
@@ -1,47 +1,68 @@
-meet requirements imposed by an external firewall device or policy.
+Assign blocks of IP addresses from an IP pool for different topological areas.
 
-There are three approaches to configuring IP address assignment behavior which
-the [IPAM section of the cni-plugin configuration reference
-document]({{site.baseurl}}/{{page.version}}/reference/cni-plugin/configuration#ipam)
-explains in detail. For the purposes of topology, IP address
-assignment must be per-host (node) which disqualifies Kubernetes annotations
-as an option since it is only configurable on a per-namespace or per-pod level.
-Left between using CNI configuration and IP pool node selectors, the
-latter is preferred as it does not require making any changes within the
-host's file system which the former does.
+### Value
 
-At a high level, node selection-based IP address assignment is exactly what it
-sounds like: node labels are set and then the appropriate node selectors
-on the desired IP pool resources are set. The remainder of this article goes
-into a detailed example of using this feature to configure IP address
-assignment based on a certain rack affinity.
+If you have workloads in different regions, zones, or rack, you may want them to get IP addresses from the same IP pool. This strategy is useful for reducing the number of routes that are required in the network, or to meet requirements imposed by an external firewall device or policy. {{site.prodname}} makes it easy to do this using an IP pool resource with node labels and node selectors.
 
-> **Important**: If Calico is unable to determine an IP pool for a workload
-> based on the above order, or if there are no IP addresses left in the
-> determined IP pools, then the workload will not be assigned an address and
-> will fail to start. To prevent this, we recommend ensuring that all nodes are
-> selected by at least one IP pool.
-{: .alert .alert-danger}
+### Features
 
-## Prerequisites
+This how-to guide uses the following {{site.prodname}} features:
 
-This feature requires {{site.prodname}} for networking in etcd mode.
+- **IPPool** resource
 
-### Example: Kubernetes
+### Concepts
 
-In this example, we created a cluster with four nodes across two racks
-(two nodes/rack). Consider the following:
+#### IP address assignment
+
+Topology-based IP address assignment requires addresses to be per-host (node). 
+As such, Kubernetes annotations cannot be used because annotations are only per-namespace and per-pod. And although you can configure IP addresses for nodes in the CNI configuration, you are making changes within the host’s file system. The best option is to use node-selection IP address assignment using IP pools.
+
+#### Node-selection IP address management
+
+Node selection-based IP address assignment is exactly what it sounds like: node labels are set, and then the appropriate node selectors on the desired IP pool resources are set. 
```

Comment:
This sentence doesn't really describe what node-selection is.

I might say "node labels are set, and Calico uses node selectors to decide which IP pools to use when assigning IP addresses to that node"

Reply:
Fixed.